src/index.tsx

```diff
@@ -29,19 +31,34 @@ export interface IReactToPrintProps {
     removeAfterPrint?: boolean;
     /** Optional - suppress error messages */
     suppressErrors?: boolean;
+    /** Optional - download pdf file directly */
+    downloadPDF?: boolean;
+    /** Optional - document title for PDF file */
+    documentTitle?: string;
 }
 
+export const templateToUri = (html) => (
+    `data:text/html;base64,${encodeURIComponent(Base64.encode(html))}`
+);
+
 export default class ReactToPrint extends React.Component<IReactToPrintProps> {
     private linkTotal: number;
     private linksLoaded: Element[];
     private linksErrored: Element[];
 
     public startPrint = (target: any, onAfterPrint: any) => {
-        const { removeAfterPrint } = this.props;
+        const { removeAfterPrint, downloadPDF, documentTitle } = this.props;
+        const docAsBase64 = templateToUri(target.contentWindow.document.documentElement.outerHTML);
```

Comment:
Looks like call to templateToUri is only needed in downloadPDF=true case and can be moved into condition on line 56